recourse_methods/mrmc_method.py

```diff
@@ -184,57 +289,143 @@ def filter_data(self, confidence_threshold: float, model: base_model.BaseModel)
+        """Generates different recourse instructions for the poi for each of
+        the k_directions.
+
+        Args:
+            poi: The Point of Interest (poi) to generate recourse instructions
+                for.
+
+        Returns:
+            A list where each element is a different recourse instruction."""
         instructions = []
         for mrm in self.mrms:
             instructions.append(mrm.get_recourse_instructions(poi))
         return instructions
 
-    def get_kth_recourse_instructions(self, poi: pd.Series, dir_index: int) -> Any:
-        """Generates a single set of recourse instructions for the kth direction."""
+    def get_kth_recourse_instructions(
+        self, poi: pd.Series, dir_index: int
+    ) -> Any:
+        """Generates a single set of recourse instructions for the kth
+        direction.
+
+        Args:
+            poi: The Point of Interest (POI) to generate recourse instructions
+                for.
+            dir_index: Which set of instructions to generate.
+
+        Returns:
+            A set of recourse instructions for the POI."""
         return self.mrms[dir_index].get_recourse_instructions(poi)
 
 
-def check_dirs(poi: recourse_adapter.EmbeddedSeries, dirs: recourse_adapter.EmbeddedDataFrame, cluster_centers: np.ndarray) -> np.ndarray:
+def check_dirs(
+    poi: recourse_adapter.EmbeddedSeries,
+    dirs: recourse_adapter.EmbeddedDataFrame,
+    cluster_centers: np.ndarray,
+) -> np.ndarray:
     """Convenience function for verifying the directions generated by MRMC.
 
-    Returns the dot product between the normalized directions and the normalized difference vector
-    between the POI and cluster centers.
-    """
+    Returns the dot product between the normalized directions and the
+    normalized difference vector between the POI and cluster centers. The
+    smaller the dot products, the more likely the directions are correct. This
+    is because MRM directions typically point towards the cluster centers.
+
+    The dirs and cluster_centers should be aligned such that the ith direction
+    corresponds to the ith cluster center.
+
+    Args:
+        poi: The Point of Interest (POI) used to generate recourse.
+        dirs: The recourse directions generated.
+        cluster_centers: An array where each row is a cluster center.
+
+    Returns:
+        An array where each element is the dot product between a recourse
+        direction and its corresponding cluster center."""
     expected_dirs = cluster_centers - poi.to_numpy()
-    expected_dirs = expected_dirs / np.linalg.norm(expected_dirs, axis=1)[:,None]
+    expected_dirs = (
+        expected_dirs / np.linalg.norm(expected_dirs, axis=1)[:, None]
+    )
     actual_dirs = dirs.to_numpy()
-    actual_dirs = actual_dirs / np.linalg.norm(actual_dirs, axis=1)[:,None]
-    return np.diag(expected_dirs @ actual_dirs.T)
\ No newline at end of file
+    actual_dirs = actual_dirs / np.linalg.norm(actual_dirs, axis=1)[:, None]
```

Comment:
Why are you indexing with None here and on line 427?

Reply:
No reason -- I've now removed the indexing.